packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1287,6 +1287,32 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
     return _of<T>(context);
   }
 
+  /// Returns the modal route most closely associated with the given context.
+  ///
+  /// Similar to [of], but this method does not cause rebuilds when the route's
+  /// state changes. This makes it more efficient for cases where you need to
+  /// read the current route's value once and don't need to rebuild when the
+  /// route changes.
+  ///
+  /// Returns null if the given context is not associated with a modal route.
+  ///
+  /// {@tool snippet}
+  /// Example usage:
+  ///
+  /// ```dart
+  /// // Read the current route's name without subscribing to changes
+  /// String? routeName = ModalRoute.readFrom(context)?.settings.name;
+  /// ```
+  /// {@end-tool}
+  ///
+  /// See also:
+  ///
+  ///  * [of], which provides the same information but will cause rebuilds when
+  ///    the route changes.
+  static ModalRoute<T>? readFrom<T extends Object?>(BuildContext context) {
+    return context.getInheritedWidgetOfExactType<_ModalScopeStatus>()?.route as ModalRoute<T>?;
```

Comment:
If we made _ModalScopeStatus public would that also solve the problem? I'm not sure whether that's a good idea, just thinking this through.

Reply:
Yes, I think making _ModalScopeStatus public would solve the problem, but i don't know reasoning behind why this was made private at first place.

Reply:
Just an idea, in #165289 I am planning to migrate all subscribing to selector pattern. So, if we start following that same pattern everywhere, we can have 2 methods for subscription and for read. WDYT ?

Reply:
I just left another review over on that PR. I'm really liking that approach but still thinking through the `Object?` types that end up all over. Maybe we should finish that PR first and then come back to this one?

Reply:
Yup, make sense.